private_set_intersection/python/psi_bindings.cpp

```diff
@@ -22,27 +22,73 @@ T throwOrReturn(const private_join_and_compute::StatusOr<T>& in) {
   return in.ValueOrDie();
 }
 
+template <class T>
+auto saveProto(const T& obj) {
+  char* output = (char*)malloc(obj.ByteSizeLong() * sizeof(char));
+  if (output == nullptr) throw std::runtime_error("out of memory");
+
+  if (!obj.SerializeToArray(output, obj.ByteSizeLong()))
+    throw std::invalid_argument("failed to serialize proto");
+  return py::bytes(output, obj.ByteSizeLong());
+}
+
+template <class T>
+auto loadProto(T& obj, const std::string& data) {
```

Comment:
Why are we saving to `py::bytes`, but loading from a `std::string`? Could this also be `py::bytes`?

Reply:
Note there is also a [`ParseFromString`](https://developers.google.com/protocol-buffers/docs/reference/cpp/google.protobuf.message_lite#MessageLite.ParseFromString.details) function in protobuf.